upass/__main__.py

```diff
@@ -274,6 +279,70 @@ def copy_selected(self, originator):
         if pathg in self.passwords:
             self.call_pass(originator, (path, True, False))
 
+    def generate_password(self, originator, args=None):
+        if args:
+            path, length, symbols, force = args
+        else:
+            path, length, symbols, force = '', 16, True, False
+        if not path and self.current != '.':
+            path = self.current + '/'
+        self.set_header('GENERATE PASSWORD')
+        self._clear_box()
+        self.generate_password_path = urwid.Edit(("highlight", "{0}{1}".format(GENERATE_PASSWORD_PROMPT, path)))
```

Comment:
I would prefer to put the path as the default text, so it can be edited by the user in case they need it.

```py
self.generate_password_path = urwid.Edit(("highlight", "Password Name: "), path)
```

(This will also help get rid of the constant. BTW, the `+` operator would look better in this case.)